build/Build.cs

```diff
@@ -184,15 +186,68 @@ public Build()
                                 nugetVersion,
                                 FixedRuntimes.Cloud);
 
-                      DoPublish(RootProjectName, Frameworks.NetCoreApp31, nugetVersion, FixedRuntimes.Portable);
-
                       // Create the self-contained Calamari packages for each runtime ID defined in Calamari.csproj
                       foreach (var rid in Solution?.GetProject(RootProjectName).GetRuntimeIdentifiers()!)
                           DoPublish(RootProjectName, Frameworks.NetCoreApp31, nugetVersion, rid);
                   });
 
+        Target PublishCalamariFlavourProjects => _ => _
+                                                          .DependsOn(Compile)
+                                                          .Executes(() =>
+                                                                    {
+                                                                        var calamariFlavourGroup = MigratedCalamariFlavours.Flavours.Select(flavour => SourceDirectory.GlobFiles($"**/{flavour}*.csproj")); //We need Calamari & Calamari.Tests
+                                                                        foreach (var calamariFlavourProjects in calamariFlavourGroup)
+                                                                        {
+                                                                            foreach (var project in calamariFlavourProjects)
+                                                                            {
+                                                                                var calamariFlavour = XmlTasks.XmlPeekSingle(project, "Project/PropertyGroup/AssemblyName");
+
+                                                                                var frameworks = XmlTasks.XmlPeekSingle(project, "Project/PropertyGroup/TargetFrameworks") ?? XmlTasks.XmlPeekSingle(project, "Project/PropertyGroup/TargetFramework");
+
+                                                                                foreach (var framework in frameworks.Split(';'))
+                                                                                {
+                                                                                    void RunPublish(string runtime, string platform)
+                                                                                    {
+                                                                                        DotNetPublish(s => s
+                                                                                                           .SetProject(project)
+                                                                                                           .SetConfiguration(Configuration)
+                                                                                                           .SetFramework(framework)
```

Comment:
In the calls below to `RunPublish`, we're doing things like:

`RunPublish("win-x64","win-x64") // for net5.0 windows`
`RunPublish("linux-x64","linux-x64") // for net5.0 linux`
`RunPublish("", "netfx") // for net452`

This tells me there's something not quite right with the parameters to this method. The mixed use of some variables from the outter scope and some as parameters is a little messy/confusing.

Reply:
I've refactored this method, the original code came directly from the [Sashimi.AzureAppService publish](https://github.com/OctopusDeploy/Sashimi.AzureAppService/blob/d5e9e0e8ccb3daf44f241e232b68d014c06392e3/build/Build.cs#L111).